cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MetricsMessageCallback.java

```diff
@@ -44,21 +46,22 @@
 public final class MetricsMessageCallback implements KafkaConsumer.MessageCallback {
 
   private static final Logger LOG = LoggerFactory.getLogger(MetricsMessageCallback.class);
-  private static final ImmutableMap<String, String> METRICS_PROCESSOR_CONTEXT =
-    ImmutableMap.of(Constants.Metrics.Tag.NAMESPACE, Constants.SYSTEM_NAMESPACE,
-                    Constants.Metrics.Tag.COMPONENT, "metrics.processor");
 
   private final DatumReader<MetricValues> recordReader;
   private final Schema recordSchema;
   private long recordProcessed;
-  private MetricStore metricStore;
+  private final MetricStore metricStore;
+  private final Map<String, String> metricsContext;
 
   public MetricsMessageCallback(DatumReader<MetricValues> recordReader,
                                 Schema recordSchema,
-                                MetricStore metricStore) {
+                                MetricStore metricStore,
+                                @Nullable
+                                MetricsContext metricsContext) {
     this.recordReader = recordReader;
     this.recordSchema = recordSchema;
     this.metricStore = metricStore;
+    this.metricsContext = metricsContext == null ? Collections.<String, String>emptyMap() : metricsContext.getTags();
```

Comment:
what does an empty map mean, they don't have any context? , then it becomes inefficient to search/query them right ?